### DISCUSSION TITLE
Position coords are updated when animation finish

### DISCUSSION THREAD

**Discussion by carmelocatalfamo** [Annoraaq/grid-engine#441] [Q&A] [answered]
HI!
I'm developing a multiplayer game with phaser 3 and grid-engine but I'm new to game development so my approach might be wrong.

In the `update` method I move player then I send position to server

```js
if (cursors.leftIsDown) {
  this.gridEngine.move(sessionId, Direction.LEFT)
} else if (cursors.rightIsDown) {
  this.gridEngine.move(sessionId, Direction.RIGHT)
} else if (cursors.upIsDown) {
  this.gridEngine.move(sessionId, Direction.UP)
} else if (cursors.downIsDown) {
  this.gridEngine.move(sessionId, Direction.DOWN)
}

const position = this.gridEngine.getPosition(sessionId)

this.multiplayer.room.send('move', {
  // ... other data
  position: enginePosition,
})
```

When other clients receive new player position I call `moveTo` method but I can see the delay on movement

```js
this.gridEngine.moveTo(sessionId, position)
```

This is because getPosition return updated value when animation finish. There is an option that i can use to get new position while player is moving to next tile?

In this video I'm moving player on left screen

https://github.com/Annoraaq/grid-engine/assets/11672358/18f2f228-76b7-45b9-aa6d-03d4910d1e90

**Comment by carmelocatalfamo** [comment, marked as answer]
I achieved that with `this.gridEngine.positionChangeStarted` method 😅

**Reply by Annoraaq** [reply]
I'm glad you found a solution :)
